solr/solrj/src/java/org/apache/solr/common/cloud/Replica.java

```diff
@@ -153,10 +152,12 @@ public boolean equals(Object o) {
 
     return name.equals(replica.name);
   }
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented Replica.hashCode()");
-//  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, nodeName, collection);
```

Comment:
`equals()` does not compare `nodeName` and `collection`, so `hashCode()` should not be based on these values.

Reply:
I'll use just name.